### ISSUE TITLE
Caching in 0.10.0-rc1 is returning wrong values

### ISSUE THREAD

**Issue by gabrielpage** [druid-io/druid#4093]
This issue is not present in 0.9.2

This can be replicated with the wikiticker quickstart. Once this data is loaded, execute the following 2 queries :

```json
{
    "queryType": "timeseries",
    "dataSource": "wikiticker",
    "granularity": "all",
    "aggregations": [
        {"type": "longSum", "name": "count", "fieldName": "count"},
        {"type": "longSum", "name": "added", "fieldName": "added"}
    ],
    "intervals": [ "2016/2018" ]
}
```

This return the correct values

```json
[ {
  "timestamp" : "2016-06-27T00:00:11.080Z",
  "result" : {
    "count" : 11774265,
    "added" : 24433
  }
} ]
```

Then execute the following query (it's identical other than the order of the aggregations)

```json
{
    "queryType": "timeseries",
    "dataSource": "wikiticker",
    "granularity": "all",
    "aggregations": [
        {"type": "longSum", "name": "added", "fieldName": "added"},
        {"type": "longSum", "name": "count", "fieldName": "count"}
    ],
    "intervals": [ "2016/2018" ]
}
```

Yields the following.

```json
[ {
  "timestamp" : "2016-06-27T00:00:11.080Z",
  "result" : {
    "count" : 24433,
    "added" : 11774265
  }
} ]
```

The values returned have been "switched". With "count" incorrectly showing the "added" value in the result, and vice versa.

Disabling caching on both the historical and the broker removes the problem. If caching is on either the broker or the historical then the error returns.

**Comment by gianm** [comment]
Thanks for the report @gabrielpage. Confirmed locally, working on a fix.

**Comment by gianm** [comment]
Patch in #4094.

**Comment by gianm** [comment]
Fixed by #4094.
